atspi-proxies/src/collection.rs

```diff
@@ -1,60 +1,95 @@
-//! section of the zbus documentation.
+//! See the documentation on the individual methods for details:
 //!
-#![allow(clippy::too_many_arguments)]
-// this allow zbus to change the number of parameters in a function without setting off clippy
+//! * [`get_matches`](struct.CollectionProxy.html#method.get_matches)
+//! * [`get_matches_from`](struct.CollectionProxy.html#method.get_matches_from)
+//! * [`get_matches_to`](struct.CollectionProxy.html#method.get_matches_to)
+//!
+//! [`CollectionProxy`]: crate::collection::CollectionProxy
 
-use crate::common::{MatchArgs, ObjectRef, SortOrder, TreeTraversalType};
+use crate::common::{ObjectMatchRule, ObjectRef, SortOrder, TreeTraversalType};
 
 #[zbus::proxy(interface = "org.a11y.atspi.Collection", assume_defaults = true)]
 trait Collection {
-	/// GetActiveDescendant method
-	fn get_active_descendant(&self) -> zbus::Result<ObjectRef>;
+	// The active descendant of the given object.
+	//
+	// Looks like this is unimplemented.
+	//
+	// See [atspi/collection.c](https://gitlab.gnome.org/GNOME/at-spi2-core/-/blob/main/atspi/atspi-collection.c?ref_type=heads#L272)
+	//
+	// fn get_active_descendant(&self) -> zbus::Result<ObjectRef>;
 
-	/* ROLE fields:
-	  &[i32]: AtspiStateSet,
-	  i32: AtspiCollectionMatchType,
-	  HashMap<&str, &str>: attributes,
-	  i32: AtspiCollectionMatchType (attribute match type),
-	  &[i32]: roles,
-	  i32: AtspiCollectionMatchType (role match type),
-	  &[&str]: interfaces,
-	  i32: AtspiCollectionMatchType (interface match type),
-	  bool: invert
-	*/
-	/// GetMatches method
+	/// Retrieves a list of objects that match the specified ObjectMatchRule, ordered according to SortOrder and limited by the count parameter.
+	///
+	/// # Arguments
+	///
+	/// * `rule` - An [`ObjectMatchRule`] describing the match criteria.
+	/// * `sortby` - A [`SortOrder`] specifying the way the results are to be sorted.
+	/// * `count` - The maximum number of results to return, or 0 for no limit.
+	/// * `traverse` - Not supported.
+	///
+	/// [`ObjectMatchRule`]: atspi_common::object_match::ObjectMatchRule
+	/// [`SortOrder`]: atspi_common::SortOrder
 	fn get_matches(
 		&self,
-		rule: &MatchArgs<'_>,
+		rule: ObjectMatchRule,
 		sortby: SortOrder,
 		count: i32,
 		traverse: bool,
 	) -> zbus::Result<Vec<ObjectRef>>;
 
-	/// GetMatchesFrom method
+	/// Retrieves objects from the collection, after `current_object`, matching a given `rule`.
+	///
+	/// # Arguments
+	///
+	/// * `current_object` - The object at which to start searching.
+	/// * `rule` - An [`ObjectMatchRule`] describing the match criteria.
+	/// * `sortby` - A [`SortOrder`] specifying the way the results are to be sorted.
+	/// * `tree` - A [`TreeTraversalType`] specifying restrictions on the objects to be traversed.
+	/// * `limit_scope` - If `true`, only descendants of `current_object`'s parent will be returned. Otherwise (if `false`), any accessible may be returned if it would preceed `current_object` in a flattened hierarchy.
```

Comment:
There's no `limit_scope` value accepted by this method. Why is it in the docs?

Reply:
Accidentally swapped some around. 
Now fixed!